### PR TITLE
Update cli11 and micromamba

### PR DESCRIPTION
**Pull request by cbourjau**
This is my first PR to this project. Please let me know if more information is required!

##### Micromamba
Update of micromamba only contains bugfixes and improvements: https://github.com/mamba-org/mamba/blob/master/CHANGELOG.md

The build dependencies now include `python3`, `tl-expected` and `cli11>=2.2`. Furthermore, micromamba is now also build for `darwin` platforms (seems to have also been fixed in https://github.com/NixOS/nixpkgs/pull/178801 )

##### Cli11
Major release needed for the latest `micromamba` release. Breaking changes are few, though: https://github.com/CLIUtils/CLI11/blob/main/CHANGELOG.md

##### Ethminer
Ethminer seems to be the only other package depending on cli11. Unfortunately, I could not test-build it since I am stuck on MacOS at the moment.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/micromamba/default.nix [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
       "-DENABLE_CONDA=true"
     ];
-
-    patches = [
-      # Patch added by the mamba team
-      (fetchpatch {
-        url = "https://raw.githubusercontent.com/mamba-org/boa-forge/20530f80e2e15012078d058803b6e2c75ed54224/libsolv/conda_variant_priorization.patch";
-        sha256 = "1iic0yx7h8s662hi2jqx68w5kpyrab4fr017vxd4wyxb6wyk35dd";
-      })
-    ];
```
Comment (mausch) [inline pkgs/tools/package-management/micromamba/default.nix:17]: Has this patch been removed/deprecated? Does it break the build with the latest changes?

Comment (mausch) [reply pkgs/tools/package-management/micromamba/default.nix:17]: The patch is still there in master : https://github.com/mamba-org/boa-forge/blob/master/libsolv/conda_variant_priorization.patch
Commit message doesn't say much https://github.com/mamba-org/boa-forge/commit/20530f80e2e15012078d058803b6e2c75ed54224 :roll_eyes: 

If it doesn't break the build let's keep this patch, if only just to keep parity with the official binaries of micromamba.

File: pkgs/tools/package-management/micromamba/default.nix [hunk 2]
```diff
@@ -66,6 +66,7 @@ stdenv.mkDerivation rec {
     description = "Reimplementation of the conda package manager";
     homepage = "https://github.com/mamba-org/mamba";
     license = licenses.bsd3;
+    platforms = platforms.all;
```
Comment (mausch) [inline pkgs/tools/package-management/micromamba/default.nix:69]: This bit has just been merged to master https://github.com/NixOS/nixpkgs/pull/178801/commits/fcda800ae2a95fb10c432ce715030d6a6e8fd402
so let's remove it from this PR.

**Comment by mausch** [comment]
After https://github.com/NixOS/nixpkgs/pull/178807#discussion_r906109568 please squash and I think it'll be ready to be merged :+1:

**Comment by cbourjau** [comment]
@mausch Squashed (into two commits) and rebased on master. Thanks for the review!

**Comment by cbourjau** [comment]
It appears that `catch2` is needed to build the tests. If it is not found, it is being downloaded during the build. I suppose it built on MacOS due to a lack of build isolation?

**Comment by mausch** [comment]
Great, all tests pass now, ran it locally just in case too.
Just squash that last commit into the first cli11 commit and we're good to go :+1:

File: pkgs/development/tools/misc/cli11/default.nix [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   lib, stdenv,
   fetchFromGitHub,
+  catch2,
   cmake,
   gtest,
   python3,
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/cli11/default.nix:7]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, catch2
, cmake
, gtest
, python3
```
and so on

File: pkgs/development/tools/misc/cli11/default.nix [hunk 4]
```diff
@@ -9,27 +10,21 @@
...
   src = fetchFromGitHub {
     owner = "CLIUtils";
     repo = "CLI11";
     rev = "v${version}";
-    sha256 = "0hbch0vk8irgmiaxnfqlqys65v1770rxxdfn3d23m2vqyjh0j9l6";
+    sha256 = "sha256-emTIaoUyTINbAAn9tw1r3zLTQt58N8A1zoP+0y41yKo=";
   };
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake catch2 ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/cli11/default.nix:22]: You are sure about this?

Comment (cbourjau) [reply pkgs/development/tools/misc/cli11/default.nix:22]: Catch2 is needed to build the tests. I am a nix beginner so i can't say that I am sure where to put such dependencies.

File: pkgs/development/tools/misc/cli11/default.nix (hunk 3 above)
Comment (cbourjau) [reply pkgs/development/tools/misc/cli11/default.nix:7]: Should i run nixpkgs-fmt on the files while touching them or is that considered unnecessary noise in the pr?

File: pkgs/development/tools/misc/cli11/default.nix (hunk 4 above)
Comment (cbourjau) [reply pkgs/development/tools/misc/cli11/default.nix]: I moved `catch2` into `checkInputs` which seems to be the correct place?

**Comment by cbourjau** [comment]
I ran `nixpkgs-fmt` on the two files which re-formatted the inputs. Furthermore, I moved the test-dependency of cli11 into the `checkInputs` which seems to be the right place for that? Lastly, everything is tidied up in two commits. Good to go @mausch ?

**Pull request merged**